File: 1614-Maximum-Nesting-Depth-of-the-Parentheses.py
class Solution(object):
    def maxDepth(self, s):
        maxDepth = 0
        currDepth = 0

        for char in s:
            if char == "(":
                currDepth += 1
                maxDepth = max(maxDepth, currDepth)
            if char == ")":
                currDepth -= 1
        
        return maxDepth

File: 1614-Maximum-Nesting-Depth-of-the-Parentheses.js
var maxDepth = function(s) {
    let maxDepth = 0;
    let currDepth = 0;
    
    for (let i = 0; i < s.length; i++) {
        let char = s[i];
        if (char === "(") {
            currDepth++;
            maxDepth = Math.max(maxDepth, currDepth);
        }
        if (char === ")") {
            currDepth--;
        }
    }
    
    return maxDepth;
};